README.md

```diff
@@ -68,926 +68,46 @@ Alternatively, packages can also be downloaded using Package Manager Console:
-Another is asynchronous execution of the query by the server (you can recognize it by `InAsyncMode` containing method names, e. g. `ExecuteInAsyncMode`, `ExecuteAsyncInAsyncMode`).
-
-Example of synchronous code starting a query to be executed asynchronously on the server:
-```cs
-using (SnowflakeDbConnection conn = new SnowflakeDbConnection("account=testaccount;username=testusername;password=testpassword"))
-{
-      conn.Open();
-      SnowflakeDbCommand cmd = (SnowflakeDbCommand)conn.CreateCommand();
-      cmd.CommandText = "SELECT ...";
-      var queryId = cmd.ExecuteInAsyncMode();
-      // ...
-}
-```
-
-Example of asynchronous code starting a query to be executed asynchronously on the server:
-```cs
-using (SnowflakeDbConnection conn = new SnowflakeDbConnection("account=testaccount;username=testusername;password=testpassword"))
-{
-      await conn.OpenAsync(CancellationToken.None).ConfigureAwait(false);
-      SnowflakeDbCommand cmd = (SnowflakeDbCommand)conn.CreateCommand())
-      cmd.CommandText = "SELECT ...";
-      var queryId = await cmd.ExecuteAsyncInAsyncMode(CancellationToken.None).ConfigureAwait(false);
-      // ...
-}
-```
-
-You can check the status of a query executed asynchronously on the server either in synchronous code:
-```cs
-var queryStatus = cmd.GetQueryStatus(queryId);
-Assert.IsTrue(conn.IsStillRunning(queryStatus)); // assuming that the query is still running
-Assert.IsFalse(conn.IsAnError(queryStatus)); // assuming that the query has not finished with error
-```
-or the same in an asynchronous code:
-```cs
-var queryStatus = await cmd.GetQueryStatusAsync(queryId, CancellationToken.None).ConfigureAwait(false);
-Assert.IsTrue(conn.IsStillRunning(queryStatus)); // assuming that the query is still running
-Assert.IsFalse(conn.IsAnError(queryStatus)); // assuming that the query has not finished with error
-```
-
-The following example shows how to get query results.
-The operation will repeatedly check the query status until the query is completed or timeout happened or reaching the maximum number of attempts.
-The synchronous code example:
-```cs
-DbDataReader reader = cmd.GetResultsFromQueryId(queryId);
-```
-and the asynchronous code example:
-```cs
-DbDataReader reader = await cmd.GetResultsFromQueryIdAsync(queryId, CancellationToken.None).ConfigureAwait(false);
-```
-
-**Note**: GET/PUT operations are currently not enabled for asynchronous executions.
-
-## Executing a Batch of SQL Statements (Multi-Statement Support)
+Connection pooling description: [Multiple Connection Pools](doc/ConnectionPooling.md).
 
-With version 2.0.18 and later of the .NET connector, you can send
-a batch of SQL statements, separated by semicolons,
-to be executed in a single request.
+Backward compatible pooling prior to v4.0.0 is described: [Single Connection Pool](doc/ConnectionPoolingDeprecated.md) - `deprecated`
```

Comment:
maybe just: Pooling prior to ....
because it can suggest there is a backward compatibility with previous pool